sljit_src/sljitProtExecAllocator.c

```diff
@@ -204,6 +203,36 @@ static SLJIT_INLINE struct chunk_header* alloc_chunk(sljit_uw size)
 	retval->fd = fd;
 	return retval;
 }
+#else
+static SLJIT_INLINE struct chunk_header* alloc_chunk(sljit_uw size)
+{
+	struct chunk_header *retval;
+	void *maprx;
+
+	retval = (struct chunk_header *)mmap(NULL, size,
+			PROT_MPROTECT(PROT_EXEC|PROT_WRITE|PROT_READ),
```

Comment:
Can you set all those flags in a secure environment?

Reply:
this call is not asking for those permissions yet, indeed the mmap will return 
a page without any permissions and those will be set by mprotect below.

PROT_MPROTECT is specially designed for NetBSD's secure environment (available since version 8) and indicates an "intention", it is also NetBSD specific and documented for this specific use on its man pages for man and mremap